protoc_plugin/test/unittest_proto3_test.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:test/test.dart';
+import '../out/protos/google/protobuf/unittest_proto3.pb.dart';
+
+void main() {
+  group('Optional fields should be nullable for', () {
+    test('int32', () {
+      final obj = TestAllTypes.create()
+        ..singleInt32 = 1
+        ..optionalSingleInt32 = 2;
+      expect(obj.singleInt32, 1);
+      expect(obj.optionalSingleInt32, 2);
+      expect(obj.hasOptionalSingleInt32(), true);
+
+      final obj2 = TestAllTypes.create()..singleInt32 = 1;
+      expect(obj2.singleInt32, 1);
+      expect(obj2.optionalSingleInt32, null);
+      // should not generate linting errors
+      expect(obj2.optionalSingleInt32 ?? 1, 1);
```

Comment:
And this line throws a linting error as the type of `obj2.optionalSingleInt32` is `int`, not `int?`.

I'm also now realizing this line should be more like `expect(obj2.optionalSingleInt32 is int?)`.